File: Mobile/Mafqoodi-Mobile/lib/domain/entities/report.dart
class Report {
  const Report({
    required this.id,
    required this.title,
    required this.description,
    required this.locationName,
    required this.reportType,
    this.latitude,
    this.longitude,
    this.category,
    this.customCategoryName,
    this.rewardAmount,
    this.rewardCurrency,
    this.imageData,
    this.publisherPhone,
    this.adminStatus,
    this.createdAt,
    required this.status,
  });

  final String id;
  final String title;
  final String description;
  final String locationName;
  final String reportType;
  final double? latitude;
  final double? longitude;
  final String? category;
  final String? customCategoryName;
  final double? rewardAmount;
  final String? rewardCurrency;
  final String? imageData;
  final String? publisherPhone;
  final String? adminStatus;
  final DateTime? createdAt;
  final String status;

  factory Report.fromJson(Map<String, dynamic> json) => Report(
    id: json['id'].toString(),
    title: json['title'] ?? '',
    description: json['description'] ?? '',
    locationName: json['locationName'] ?? '',
    reportType: json['reportType'] ?? 'lost',
    latitude: (json['latitude'] as num?)?.toDouble(),
    longitude: (json['longitude'] as num?)?.toDouble(),
    category: json['category'],
    customCategoryName: json['customCategoryName'],
    rewardAmount: (json['rewardAmount'] as num?)?.toDouble(),
    rewardCurrency: json['rewardCurrency'],
    imageData: json['imageData'],
    publisherPhone: json['publisherPhone'],
    adminStatus: json['adminStatus'],
    createdAt: DateTime.tryParse(json['createdAt']?.toString() ?? ''),
    status: json['status'] ?? 'active',
  );
}

File: Backend/Mafqoodi.Domain/Entities/Report.cs
namespace Mafqoodi.Domain.Entities;

public sealed class Report
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public string LocationName { get; set; } = string.Empty;
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public string? PublisherPhone { get; set; }
    public string PublisherAccountType { get; set; } = "personal";
    public required string ReportType { get; set; }
    public string? Category { get; set; }
    public string? CustomCategoryName { get; set; }
    public decimal? RewardAmount { get; set; }
    public string? RewardCurrency { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string Status { get; set; } = "active";
    public string AdminStatus { get; set; } = "normal";
    public string? ImageData { get; set; }
    public string? Base64Image { get; set; }
    public ICollection<ReportFlag> Flags { get; set; } = new List<ReportFlag>();
}

public sealed class ReportFlag
{
    public Guid Id { get; set; }
    public Guid ReportId { get; set; }
    public Report? Report { get; set; }
    public required Guid UserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
